File: dongyimai_cart_interface/src/main/java/com/offcn/cart/service/CartService.java
package com.offcn.cart.service;

import com.wzp.entity.Cart;

import java.util.List;

/**
 * 购物车服务接口
 *
 * @author Administrator
 */
public interface CartService {

    //添加购物车
    public List<Cart> addGoodsToCartList(List<Cart> srccartList, Long itemId, Integer num);

    //根据商家名称在缓存中发现购物车
    public List<Cart> findCartListFromRedis(String username);

    //根据商家名称保存购物车列表
    public void saveCartListRedis(List<Cart> cartList, String username);

    //合并购物车
    public List<Cart> margeCartList(List<Cart>cartList_cookie,List<Cart>cartList_redis);

}

File: dongyimai_cart_web/src/main/webapp/js/service/cartService.js
app.service('cartService', function ($http) {
    this.findCartList = function () {
        return $http.get('../cart/findCartList.do');
    }

    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('../cart/addGoodsToCartList.do?itemId=' + itemId + '&num=' + num);
    }
    //获取总金额和总件数
    this.sum = function (cartList) {
        var totalValue = {'totalNum': 0, 'totalMoney': 0.00};
        for (var i = 0; i < cartList.length; i++) {
            var orderItemList = cartList[i].orderItemList;
            for (var j = 0; j < orderItemList.length; j++) {
                totalValue.totalNum += orderItemList[j].num;
                totalValue.totalMoney += orderItemList[j].totalFee;   //总金额
            }
        }
        return totalValue;
    }

    //提交订单
    this.submitOrder = function (entity) {
        return $http.post('../order/add.do', entity);
    }
})